services/workflows-service/src/workflow/workflow-event-emitter.service.ts

```diff
@@ -2,21 +2,29 @@ import { EventEmitter2 } from '@nestjs/event-emitter';
 import { Injectable } from '@nestjs/common';
 import { ExtractWorkflowEventData, TEventName } from '@/workflow/types';
 
+export type EventConfig = {
+  forceEmit?: boolean;
+};
+
 @Injectable()
 export class WorkflowEventEmitterService {
   constructor(private eventEmitter: EventEmitter2) {}
 
-  emit<TEvent extends TEventName>(eventName: TEvent, eventData: ExtractWorkflowEventData<TEvent>) {
+  emit<TEvent extends TEventName>(
+    eventName: TEvent,
+    eventData: ExtractWorkflowEventData<TEvent>,
+    config = { forceEmit: false },
+  ) {
     if (!eventName) {
```

Comment:
in case someone will use this config with a partial flags then forceEmit wont be present
extract { forceEmit: false } to a default contant and use merge config with default
```
const DEFAULT_CONFIG = { forceEmit: false };
  emit<TEvent extends TEventName>(
    eventName: TEvent,
    eventData: ExtractWorkflowEventData<TEvent>,
    config = { forceEmit: false },
  )
    config = {...DEFAULT_CONFIG, ...(config || {}) }
  ```

Reply:
Done